### DISCUSSION TITLE
Deployment to ECS via Argo-cd

### DISCUSSION THREAD

**Discussion by kunal-parsewar** [argoproj/argo-cd#17843] [Q&A] [answered]
Hi Team,

For the moment we can use argo to deploy the code to EKS / K8s Clusters. Do we have any provision in ArgoCd to deploy the code to ECS cluster similar to EKS / K8s one?

**Comment by csantanapr** [comment, marked as answer]
ArgoCD only support the deployment to Kubernetes Resources.
For Amazon ECS you would need ArgoCD to deploy a Custom Resource for ECS to Kubernetes (etcd), then a Kubernetes controller like ACK or Crossplane will be able to reconcile from etcd to the Cloud provider (AWS)
Here are the docs:
**ACK:**
  - https://aws-controllers-k8s.github.io/community/docs/community/how-it-works/
  - https://aws-controllers-k8s.github.io/community/docs/community/services/#amazon-ecs
  - https://aws-controllers-k8s.github.io/community/reference/ecs/v1alpha1/service/

**Crossplane:**
  - https://docs.crossplane.io/v1.16/getting-started/introduction/
  - https://github.com/crossplane-contrib/provider-upjet-aws
  - https://github.com/crossplane-contrib/provider-upjet-aws/blob/main/examples/ecs/v1beta1/service.yaml
